nimble/host/src/ble_sm_sc.c

```diff
@@ -612,6 +612,14 @@ ble_sm_sc_public_key_rx(uint16_t conn_handle, struct os_mbuf **om,
     }
 
     cmd = (struct ble_sm_public_key *)(*om)->om_data;
+    /* Check if the peer public key is same as our generated public key.
+     * Return fail if the public keys match. */
+    if (ble_sm_sc_keys_generated) {
```

Comment:
I think this check is not needed as ble_sm_sc_ensure_keys_generated() is called before

Reply:
Yes, the check is not needed.